### PR TITLE
Implement basic keybind Labels

### PR DESCRIPTION
**Pull request by Zeffuro**
~~Don't merge yet, I need help with 2 tiny things.~~

~~1. These keybinds currently only show up when a cooldown or item is on cooldown.~~ Was a skill issue
~~2. I'd like to move `keybindHelper.UpdateKeybindHints();` to a better place as per my comment:~~ Changed with the new jobchange event + on login/plugin start
~~// TODO: Needs to be moved to some place/event that only gets triggered on either job swap or when someone updates their hotbar~~
~~// AgentUpdateFlags ActionBarUpdate is no good because that will trigger by using actions as well apparently~~

If these 2 things are solved then it'd be good to go/merge. Hope anyone can assist me on this.

Things to think about:
- ~~Do we want a generic toUpper and toLower method (different people prefer different visual styles, I highly prefer sQ for Shift Q for example but some might want SQ).~~ Added
- ~~Any replacements I missed for the formatted versions?~~ Haven't found modifiers I have missed.

Edit: For some reason I can't assign reviewers, @Tischel @WesBosch

### CODE REVIEW

**Comment by Zeffuro** [comment]
Actually, option 1 works fine, I just didn't notice/realize that the default label adds a condition...
So it should be good to use/merge, 2 is more of a neat/performance thing.

**Comment by Zeffuro** [comment]
Waiting for new Dalamud release I suppose that has job change event built in so I can move the keybind helper update on job change there and then it should be more performant too

**Comment by Zeffuro** [comment]
This feature should be done and good to go/merge, for some reason the JobChange event wasn't merged/released with the last version of Dalamud so for now it only works on Staging but it works perfectly fine there.

We can merge when JobChange hits Dalamud Release

**Comment by Zeffuro** [comment]
Changes are in live Dalamud.
@Tischel Can you look at this and merge if you agree with the PR :)

**Comment by Zeffuro** [comment]
Made one small tweak and the functionality still works fine.
@Tischel Can you check and if you happen to not want this functionality or just find me code to be too bad you're free to just close it, this PR has been in limbo for over 4 months at this point.

File: DelvCD/Helpers/KeybindHelper.cs [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+        GC.SuppressFinalize(this);
+    }
+
+    protected void Dispose(bool disposing)
+    {
+        if (!disposing)
+        {
+            return;
+        }
+
```
Comment (Tischel) [inline DelvCD/Helpers/KeybindHelper.cs:42]:
```suggestion

Singletons.Get<IClientState>().ClassJobChanged -= OnJobChanged;

```

File: DelvCD/PluginManager.cs [hunk 2]
```diff
@@ -72,6 +72,7 @@ public PluginManager(
             );
 
             ClientState.Logout += OnLogout;
+            ClientState.Login += OnLogin;
```
Comment (Tischel) [inline DelvCD/PluginManager.cs:75]: Unsubscribe from this in the dispose method

File: DelvCD/Helpers/KeybindHelper.cs [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+    }
+    
+    #endregion
+    
+    private readonly Dictionary<uint, string> _keybindActionHints = new();
+    private readonly Dictionary<uint, string> _keybindItemHints = new();
+    private readonly string?[] _actionBars = ["_ActionBar09", "_ActionBar", "_ActionBar01", "_ActionBar02", "_ActionBar03", "_ActionBar04", "_ActionBar05", "_ActionBar06", "_ActionBar07", "_ActionBar08"];
+    
+    private void ActionBarUpdateRequested(AddonActionBarX* addon) {
+        var barId = addon->AddonActionBarBase.RaptureHotbarId;
```
Comment (Tischel) [inline DelvCD/Helpers/KeybindHelper.cs:53]: Please use explicit types instead of `var` through out this file.

**Review by Tischel — CHANGES_REQUESTED** [review]

**Comment by Zeffuro** [comment]
Thank you, made all the changes!

**Pull request merged**